umqtt.simple/umqtt/simple.py

```diff
@@ -43,9 +45,15 @@ def connect(self, clean_session=True):
         msg = bytearray(b"\x10\0\0\x04MQTT\x04\x02\0\0")
         msg[1] = 10 + 2 + len(self.client_id)
         msg[9] = clean_session << 1
+        if self.user and self.pswd:
```

Comment:
What if user and/or password is (intentionally) the empty string?  Is that allowed by the spec?  If so, this check here should be `if self.user is not None and self.pswd is not None:` (or it could be "optimised" to `if self.user is not None:`, or even better to `if self.user:` if the spec doesn't allow empty user names).